server/src/main/java/org/opensearch/search/profile/aggregation/AggregationProfiler.java

```diff
@@ -47,29 +48,185 @@
+                );
+                maxSliceEndTimeMap.put(
+                    breakdownTimingType,
+                    Math.max(maxSliceEndTimeMap.getOrDefault(breakdownTimingType, Long.MIN_VALUE), endTime)
+                );
+            }
+
+            // Profiled breakdown count stats
+            for (String breakdownCountType : breakdownCountStatsTypes) {
+                buildBreakdownStatsMap(countStatsMap, profileResult, breakdownCountType);
+            }
+
+            // Profiled breakdown count
+            for (AggregationTimingType timingType : AggregationTimingType.values()) {
+                String breakdownType = timingType.toString();
+                String breakdownTypeCount = breakdownType + "_count";
+                breakdown.put(
+                    breakdownTypeCount,
+                    breakdown.getOrDefault(breakdownTypeCount, 0L) + profileResult.getTimeBreakdown().get(breakdownTypeCount)
+                );
+            }
+
+            debug = profileResult.getDebugInfo();
+            children.addAll(profileResult.getProfiledChildren());
+        }
+        // nodeTime
+        long nodeTime = maxSliceNodeEndTime - minSliceNodeStartTime;
+        avgSliceNodeTime /= tree.size();
+
+        // Profiled breakdown time stats
+        for (AggregationTimingType breakdownTimingType : AggregationTimingType.values()) {
+            buildBreakdownMap(tree, breakdown, timeStatsMap, breakdownTimingType.toString());
+        }
+
+        // Profiled breakdown total time
+        for (AggregationTimingType breakdownTimingType : AggregationTimingType.values()) {
+            String breakdownType = breakdownTimingType.toString();
+            breakdown.put(breakdownType, maxSliceEndTimeMap.get(breakdownType) - minSliceStartTimeMap.get(breakdownType));
+        }
+
+        // Profiled breakdown count stats
+        for (String breakdownCountType : breakdownCountStatsTypes) {
+            buildBreakdownMap(tree, breakdown, countStatsMap, breakdownCountType);
+        }
+
+        // children
+        List<ProfileResult> reducedChildrenTree = new LinkedList<>();
+        if (!children.isEmpty()) {
+            Map<String, List<ProfileResult>> sliceLevelAggregationMap = getSliceLevelAggregationMap(children);
+            for (List<ProfileResult> profileResults : sliceLevelAggregationMap.values()) {
+                reducedChildrenTree.addAll(reduceProfileResultsTree(profileResults));
+            }
+        }
+
+        ProfileResult reducedResult = new ProfileResult(
+            type,
+            description,
+            breakdown,
+            debug,
+            nodeTime,
+            maxSliceNodeTime,
+            minSliceNodeTime,
+            avgSliceNodeTime,
+            reducedChildrenTree
+        );
+        List<ProfileResult> ret = new LinkedList<>();
+        ret.add(reducedResult);
+        return ret;
```

Comment:
You can instead use:
```suggestion
        return List.of(reducedResult);
```